packages/clay-core/src/tree-view/useMultipleSelection.tsx

```diff
@@ -106,12 +189,35 @@ export function useMultipleSelection(
 
 			const childrenKeyMap = layoutKeys.current.get(key) as LayoutInfo;
 
-			toggleChildrenSelection(childrenKeyMap, selecteds, select);
+			if (childrenKeyMap) {
+				toggleChildrenSelection(childrenKeyMap, selecteds, select);
+			} else {
+				const parentKeyMatch = key.toString().match(/^(\$\.)(\d)/);
+				let parentKey;
+				if (parentKeyMatch && parentKeyMatch.length === 3) {
+					parentKey = `$.${parentKeyMatch[2]}`;
+				}
+				createPartialLayoutItem(key, parentKey);
+			}
```

Comment:
I think this implementation would not be necessary for the parent because since we are clicking on a Node it means that the parent Nodes are also rendered and we have them listed in the `layoutKeys`.

Reply:
You're right, I actually saw that it wasn't necessary; the only that I didn't like about that is that most of the time we call `createPartialLayoutItem` with two arguments for children (`key` and `parentKey`) so it felt a bit strange calling the same function with only one argument; but since it works and allows us to get rid of that  unnecessary check, I agree that it's better.